test/jest.setup.js

```diff
@@ -63,6 +63,7 @@ global.expectToThrowCustomError = async (func, code, words, expectedErrorDetails
 
 global.expectToThrowBadArg = async (received, words, expectedErrorDetails) => global.expectToThrowCustomError(received, 'ERROR_BAD_ARGUMENT', words, expectedErrorDetails)
 global.expectToThrowForbidden = async (received, expectedErrorDetails) => global.expectToThrowCustomError(received, 'ERROR_BAD_CREDENTIALS', ['cannot', 'access', 'credentials'], expectedErrorDetails)
+global.expectToThrowFirewall = async (received, expectedErrorDetails) => global.expectToThrowCustomError(received, 'ERROR_FIREWALL', ['your', 'IP', 'blocked', 'firewall'], expectedErrorDetails)
```

Comment:
These convenience methods are much harder to understand than just writing out the tests.

Reply:
yes true, I will need to refactor this eventually
thanks for the review